corehq/apps/app_manager/static/app_manager/js/releases/app_view_release_manager.js

```diff
@@ -43,6 +43,62 @@ hqDefine("app_manager/js/releases/app_view_release_manager", function () {
         $profilesTab.koApplyBindings(profileManager);
     }
 
+    var appReleaseLogsModel = function () {
+        let self = {};
+        self.releaseLogs = ko.observableArray();
+        self.fetchLimit = ko.observable();
+        self.totalItems = ko.observable();
+        self.fetchState = ko.observable();
+
+        self.onPaginationLoad = function () {
+            self.goToPage(1);
+        };
+
+        self.goToPage = function (page) {
+            if (self.fetchState() === 'pending') {
+                return false;
+            }
+            self.fetchState('pending');
+            $.ajax({
+                url: self.getApiEndpoint("paginate_release_logs"),
+                dataType: 'json',
+                data: {
+                    page: page,
+                    limit: self.fetchLimit,
+                },
+                success: function (data) {
+                    self.releaseLogs(
+                        _.map(data.app_release_logs, function (log) {
+                            return ko.mapping.fromJS(log);
+                        })
+                    );
+                    self.totalItems(data.pagination.total);
+                    self.fetchState('');
+                },
+                error: function () {
+                    self.fetchState('error');
+                },
+            });
+        };
+
+        self.getApiEndpoint = function () {
+            for (var i = 1; i < arguments.length; i++) {
```

Comment:
what arguments is this working on? cant you directly use the registered url in view `paginate_release_logs`?

Reply:
this function takes the registered url slug i.e `paginate_release_logs` and calls the reverse function in initial_page_data which gives back the url of the endpoint.
https://github.com/dimagi/commcare-hq/blob/485a41cb902eae8acc6e5ffdf477104f49b9106c/corehq/apps/hqwebapp/static/hqwebapp/js/initial_page_data.js#L88

Reply:
Are there any arguments that are actually passed? It looks like the `page` and other params are passed as a part of the data object in the request and not in the url. Can you confirm what arguments are passed here?

Reply:
`getApiEndpoint` is a variadic function, look at the loop that is parsing the variable passed as args, i picked this function up from another file. @orangejenny suggested in a review comment above to directly use the `reverse` function which is wrapped by this function. That looks like a better way to go than having a function for a getting a single api endpoint.

Reply:
I thought it was a mistype! Thanks @pxwxnvermx !